In [3]:
import numpy as np
from scipy.sparse import diags
from scipy.linalg import eigh
from math import ceil

In [34]:
def compute(n, k=1):
    A = diags([1, 1], [-1, 1], shape=(n,n)).todense()
    A[0,2:] = A[0,2:] + 1
    A[2:,0] = A[2:,0] + 1

    D = np.diag([np.sum(i) for i in A])
    L = D - A
    
    eval_ = eigh(L, eigvals_only=True, eigvals=(k,k))[0]
    #print(eval_)
    temp = n*(n*(n**2*(eval_- 1) - np.pi**2) - 2*np.pi**2)
    print(temp)

In [41]:
for i in range(3,13):
    n = 2**i
    print('n: {}'.format(n))
    
    compute(n, 1)

n: 8
21.694682056235536
n: 16
21.7916091048711
n: 32
21.67912165716575
n: 64
21.59344013025884
n: 128
21.54432524026288
n: 256
21.518313335984203
n: 512
21.50144909142182
n: 1024
21.496686456848693
n: 2048
23.335591439405107
n: 4096
-40.6708805183589


In [42]:
21.49/(np.pi)

6.840479454089661

# $n\left(n\left(n^2(\lambda_2-1)-\pi^2\right)-2\pi^2\right) \to \approx 21.49$

In [73]:
def compute(n, k=1):
    A = diags([1, 1], [-1, 1], shape=(n,n)).todense()
    A[0,2:] = A[0,2:] + 1
    A[2:,0] = A[2:,0] + 1

    D = np.diag([np.sum(i) for i in A])
    L = D - A
    
    eval_ = eigh(L, eigvals_only=True, eigvals=(k,k))[0]
    #print(eval_)
    temp = n*(8*np.pi**2 - n*(n**2 *(eval_- 1) -4*np.pi**2))
    print(temp)

In [75]:
for i in range(3,13):
    n = 2**i
    print('n: {}'.format(n))
    
    compute(n, 2)

n: 8
73.90186517533834
n: 16
38.01449418590664
n: 32
23.69371109219992
n: 64
17.32586644627554
n: 128
14.325978546108672
n: 256
12.870327283504594
n: 512
12.155321188023663
n: 1024
11.85065651635523
n: 2048
11.704509242379572
n: 4096
48.6835220734356


In [72]:
78.95112011631136/np.pi**2

7.999420940073051

# $n\left(8\pi^2 - n\left(n^2(\lambda_2-1)-\pi^2\right)\right) \to \approx 11.7$

In [94]:
def compute(n, k=1):
    A = diags([1, 1], [-1, 1], shape=(n,n)).todense()
    A[0,2:] = A[0,2:] + 1
    A[2:,0] = A[2:,0] + 1

    D = np.diag([np.sum(i) for i in A])
    L = D - A
    
    eval_ = eigh(L, eigvals_only=True, eigvals=(k,k))[0]
    #print(eval_)
    temp = n*(n*(n*(3 - eval_) - np.pi) - np.pi)
    print(temp)

for i in range(3,12):
    n = 2**i
    print('n: {}'.format(n))
    
    compute(n, int(n/2 - 1))

In [96]:
1.8493511124761426/np.pi

0.5886667421261477

# $n\left(n\left(n(3-\lambda_{n/2})-\pi\right)-\pi\right) \to \approx 1.85$